### ISSUE TITLE
Separated waiting list based on town id

### ISSUE THREAD

**Issue by EPuncker** [otland/forgottenserver#3889]
### Explanation of what you want to do that is currently impossible
Having separated online players limit (waiting list) based on town id

### Current Situation
If you launch a very successful ot from the start, lets say 900 players at beginning, everyone in the starting no vocation map (rook or similar), it will be impossible to hunt (or walk, depending on your walk through settings)

### My Proposal
Official game uses something that I want to suggest we to have it implemented as well: they have two separated waiting lists, for example:

- rook map (town id X) have a limit of 50 online players (no vocation)
- main map (town id != X) have a limit of 1050 online players (any vocation)

that way they can control population based on map size

**Comment by marmichalski** [comment]
Is this still valid? There is a Dawnport now, too.

**Comment by Zbizu** [comment]
personally I'd just build bigger dungeons

**Comment by EPuncker** [comment]
> Is this still valid? There is a Dawnport now, too.

it is, dawnport is limited as well (even thought it is bigger):

![2022-01-08 17_51_17-alumbra tibia run 💛 O PRIMEIRO OPEN PVP SEM GB - BOOYAH!](https://user-images.githubusercontent.com/4684880/148659630-f2561b1e-89ef-41f5-8721-e643715a6877.png)

**Comment by marmichalski** [comment]
Thanks best friend

**Comment by EPuncker** [comment]
> Thanks best friend.
> 
> The old message mentioned premium, maybe there are some situations when this is still used.
> 
> > There are too many players online on the beginners' island,Rookgaard. Only players with premium accounts are admitted at the moment

indeed, it seems there are three lists and since rook have a premium area it make sense to have that premium bypass

**Comment by Zbizu** [comment]
Premium players skip the waiting list. The whole point of the waiting list is to sell premium.

**Comment by marmichalski** [comment]
> Premium players skip the waiting list. The whole point of the waiting list is to sell premium.

There was an overall record of 64k players online in tibia. Given all of them were premium and decided to login into the same game world, do you think the game would let them? 😎

**Comment by Zbizu** [comment]
> > Premium players skip the waiting list. The whole point of the waiting list is to sell premium.
> 
> There was an overall record of 64k players online in tibia. Given all of them were premium and decided to login into the same game world, do you think the game would let them? 😎

64k across all worlds. Antica was usually the biggest world. We'd have to ask regular players who joined earlier than that.

**Comment by marmichalski** [comment]
> > > Premium players skip the waiting list. The whole point of the waiting list is to sell premium.
> > 
> > 
> > There was an overall record of 64k players online in tibia. Given all of them were premium and decided to login into the same game world, do you think the game would let them? 😎
> 
> 64k across all worlds.

This is what overall record means.
Antica had a record of 1055 players and it was quite recently.

Anyway, thanks for dodging the question, please share more of your personal preferences.

**Comment by marmichalski** [comment]
> Premium players skip the waiting list. The whole point of the waiting list is to sell premium.

Premium players just had their own pool, as expected.
```
MaxClients = 950
MaxNewbies = 300
PremiumBuffer = 100
PremiumNewbieBuffer = 150
```

And when even the waiting list would get exhausted, you'd be disconnected with one of these messages:
```
.rodata:0810BF60	00000059	C	The server is overloaded.\nOnly players with premium accounts\nare admitted at the moment.
.rodata:0810C040	00000032	C	The server is not online.\nPlease try again later.
.rodata:0810D5EA	0000001A	C	The server is overloaded.
```

**Comment by nekiro** [comment]
Tibia has upper limit of 1k people per server, doesnt matter if you are premium or not, it wont let you in, but you have priority above free players

**Comment by marmichalski** [comment]
> Tibia has upper limit of 1k people per server, doesnt matter if you are premium or not, it wont let you in, but you have priority above free players

It's 1050 to be precise. But the last 100 **must** be premium.
